python/hsfs/engine/spark.py

```diff
@@ -691,15 +711,20 @@ def read_options(self, data_format, provided_options):
             options.update(provided_options)
         return options
 
-    def parse_schema_feature_group(self, dataframe):
-        return [
-            feature.Feature(
-                feat.name.lower(),
-                feat.dataType.simpleString(),
-                feat.metadata.get("description", None),
+    def parse_schema_feature_group(self, dataframe, using_hudi):
+        features = []
+        for feat in dataframe.schema:
+            name = feat.name.lower()
+            try:
+                converted_type = self.convert_spark_type(feat.dataType, using_hudi)
+            except ValueError as e:
+                raise FeatureStoreException(f"Feature '{name}': {str(e)}")
```

Comment:
Because I want to include the feature name in the message, which is not available inside of the called function. I could e.g. return None instead of raising the ValueError, if you'd prefer that.